### PR TITLE
Updating Witness Readme file

### PR DESCRIPTION
**Pull request by yogeshbdeshpande**

### CODE REVIEW

File: binary_transparency/firmware/cmd/ft_witness/README.md [hunk 1]
```diff
@@ -0,0 +1,35 @@
+# STH Witness
+
+
+## Introduction
+
+Firmware Transparency Demo steps outlined under [Firmware Transparency](firmware) provides a robust facility against verifying that the supplied firmware update by a FW Vendor is visible via a public log server. The supplied proofs from the log server and its verification locally by clients give sufficient confidence that the FW is verifiable, and the FT log tree is consistent, as presented to individual clients by the log server.
```
Comment (mhutchinson) [inline binary_transparency/firmware/cmd/ft_witness/README.md:6]: This hyperlink doesn't seem to work. I think it's trying to find it relative to this file, so you might need to start with `../../` or `/binary_transparency/firmware/` - but I'm just guessing!

Comment (mhutchinson) [inline binary_transparency/firmware/cmd/ft_witness/README.md:6]: I would change "FW is verifiable" to "FW is discoverable".

File: binary_transparency/firmware/cmd/ft_witness/README.md [hunk 2]
```diff
@@ -0,0 +1,35 @@
...
+
+However, the current mechanism does not ensure that the view presented to all the FT clients is a consistent view of the log, i. e. the server is not presenting a forked or a different view to a subset of clients. In the transparency world, this is known as split-view attack.
+
+## STH Witness Feature
```
Comment (mhutchinson) [inline binary_transparency/firmware/cmd/ft_witness/README.md:10]:
```suggestion
## Checkpoint Witness Feature
```

File: binary_transparency/firmware/cmd/ft_witness/README.md [hunk 3]
```diff
@@ -0,0 +1,35 @@
...
+
+To ensure that all clients have consistent view of the log, FT demo has been enhanced to run an independent witness server. The role of witness server is to maintain a reference log checkpoint, which can be requested by clients for verification.  To achieve this, a witness would periodically fetch the latest log checkpoint from the log server. The latest checkpoint is compared with the already saved reference log checkpoint with the witness. If different (i. e. the log server has new entries) the witness would then check the consistency of the fetched checkpoint from its previously stored reference log checkpoint to ensure that the newly received checkpoint is consistent. Upon verification the witness overwrites its own reference log checkpoint, with the newly received checkpoint. The reference log checkpoint is now made available to external clients. Reference log checkpoint of witness server is now referred as “Witness checkpoint”.
```
Comment (mhutchinson) [inline binary_transparency/firmware/cmd/ft_witness/README.md:12]:
```suggestion
To ensure that all clients have consistent view of the log, FT demo includes an independent witness server. The role of witness server is to maintain a reference log checkpoint, which can be requested by clients for verification.  To achieve this, the witness periodically fetches the latest log checkpoint from the log server. The latest checkpoint is compared with the already saved reference log checkpoint with the witness. If different (i. e. the log server has new entries) the witness then checks the fetched checkpoint with its previously stored reference log checkpoint to ensure that the newly received checkpoint is consistent. Upon verification the witness overwrites its own reference log checkpoint with the newly received checkpoint. The reference log checkpoint is now made available to external clients. Reference log checkpoint of witness server is now referred as “Witness checkpoint”.
```

File: binary_transparency/firmware/cmd/ft_witness/README.md [hunk 4]
```diff
@@ -0,0 +1,35 @@
...
+
+Any device client receiving the Firmware Update, from firmware vendor can perform additional verification steps using the Witness Server:
```
Comment (mhutchinson) [inline binary_transparency/firmware/cmd/ft_witness/README.md:14]:
```suggestion
Any device client receiving the Firmware Update from the firmware vendor can perform additional verification steps using the Witness Server:
```

File: binary_transparency/firmware/cmd/ft_witness/README.md [hunk 5]
```diff
@@ -0,0 +1,35 @@
...
+* Using CProof, client can now locally verify the consistency between the Update log checkpoint and Witness Checkpoint, to ensure that it is not presented a forked view of the FT log tree.
+
+## Example Workflow
+
+First run the Witness Server to periodically poll the log server to fetch its witness checkpoint:
+
+```bash
+# Run the witness server using the below command
+go run ./cmd/ft_witness/ --logtostderr -v=2 --ws_db_file="/tmp/ws_ft.db"
+
```
Comment (mhutchinson) [inline binary_transparency/firmware/cmd/ft_witness/README.md:27]:
```suggestion
```

File: binary_transparency/firmware/cmd/ft_witness/README.md [hunk 6]
```diff
@@ -0,0 +1,35 @@
...
+```
+
+The witness server will now be running at localhost:8020. We can point the flash tool at this server to perform additional checks by passing --witness_url=http://localhost:8020 when flashing to the device, e.g:
```
Comment (mhutchinson) [inline binary_transparency/firmware/cmd/ft_witness/README.md:30]:
```suggestion
The witness server will now be running at localhost:8020. We can point the flash tool at this server to perform additional checks by passing `--witness_url=http://localhost:8020` when flashing to the device, e.g:
```

File: binary_transparency/firmware/cmd/ft_witness/README.md [hunk 7]
```diff
@@ -0,0 +1,35 @@
+# STH Witness
+
+
```
Comment (mhutchinson) [inline binary_transparency/firmware/cmd/ft_witness/README.md:3]:
```suggestion
```

**Review by mhutchinson — APPROVED** [review]
Looks broadly good to me. I've left a few comments on how to make this flow a little clearer / be more direct.

File: binary_transparency/firmware/cmd/ft_witness/README.md (hunk 1 above)
Comment (yogeshbdeshpande) [reply binary_transparency/firmware/cmd/ft_witness/README.md:6]: Agree, done now 👍

Comment (yogeshbdeshpande) [reply binary_transparency/firmware/cmd/ft_witness/README.md:6]: Done 👍

**Comment by yogeshbdeshpande** [comment]
> @googlebot I signed it!

**Pull request merged**